modules/user-roles/module.php

```diff
@@ -0,0 +1,179 @@
+<?php
+
+// Check for Widget Context plugin
+if ( ! class_exists( 'widget_context' ) )
+    die;
+
+
+// Go!
+WidgetContextUserRole::instance();
+
+
+class WidgetContextUserRole {
+
+    private static $instance;
+    private $wc;
+
+
+    static function instance() {
+
+        if ( ! self::$instance )
+            self::$instance = new self();
+
+        return self::$instance;
+
+    }
+
+
+    private function __construct() {
+
+        $this->wc = widget_context::instance();
+
+        add_filter( 'widget_contexts', array( $this, 'add_context' ) );
+
+        add_filter( 'widget_context_control-user_roles', array( $this, 'context_controls' ), 10, 2 );
+
+        add_filter( 'widget_context_check-user_roles', array( $this, 'context_check' ), 10, 2 );
+
+    }
+
+
+    function add_context( $contexts ) {
+
+        $contexts[ 'user_roles' ] = array(
+            'label' => __( 'User Roles', 'widget-context' ),
+            'description' => __( 'Context based user roles', 'widget-context' ),
+            'weight' => 10
+        );
+
+        return $contexts;
+
+    }
+
+
+    function context_check( $check, $settings ) {
+
+        // Assume:
+        //   "return $check" = Pass-through case
+        //   "return true"   = Affirmative case
+
+        if ( empty( $settings ) ) {
+
+            return $check;
+        }
+
+        $current_user = wp_get_current_user();
+        $is_logged_in = ( $current_user->ID !== 0 && $current_user->ID );
+
+        //
+        //  1. Logged-in/logged-out check:
+        //
+
+        // Check for logged-out user
+        if( $settings['logged-out'] === '1' && $is_logged_in !== true ) {

```

Comment:
@folkhack You should probably ensure that `logged-out` exists before using it. A good solution is `wp_parse_args()` with a set of defaults.